husarion_ugv_navigation/launch/bringup_launch.py

```diff
@@ -0,0 +1,240 @@
+# limitations under the License.
+
+
+from launch import LaunchDescription
+from launch.actions import (
+    DeclareLaunchArgument,
+    GroupAction,
+    IncludeLaunchDescription,
+    SetEnvironmentVariable,
+)
+from launch.conditions import IfCondition, UnlessCondition
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch.substitutions import (
+    EnvironmentVariable,
+    LaunchConfiguration,
+    PathJoinSubstitution,
+    PythonExpression,
+)
+from launch_ros.actions import Node, PushRosNamespace
+from launch_ros.descriptions import ParameterFile
+from launch_ros.substitutions import FindPackageShare
+from nav2_common.launch import ReplaceString, RewrittenYaml
+
+
+def generate_launch_description():
+    # Get the launch directory
+    husarion_ugv_navigation = FindPackageShare("husarion_ugv_navigation")
+    launch_dir = PathJoinSubstitution([husarion_ugv_navigation, "launch"])
+
+    autostart = LaunchConfiguration("autostart")
+    log_level = LaunchConfiguration("log_level")
+    map = LaunchConfiguration("map")
+    namespace = LaunchConfiguration("namespace")
+    observation_topic = LaunchConfiguration("observation_topic")
+    observation_topic_type = LaunchConfiguration("observation_topic_type")
+    params_file = LaunchConfiguration("params_file")
+    slam = LaunchConfiguration("slam")
+    use_composition = LaunchConfiguration("use_composition")
+    use_respawn = LaunchConfiguration("use_respawn")
+    use_sim_time = LaunchConfiguration("use_sim_time")
+
+    declare_autostart_arg = DeclareLaunchArgument(
+        "autostart",
+        default_value="true",
+        description="Automatically startup the nav2 stack",
+    )
+    declare_log_level_arg = DeclareLaunchArgument(
+        "log_level", default_value="info", description="log level"
+    )
+    declare_map_arg = DeclareLaunchArgument(
+        "map", default_value="/maps/map.yaml", description="Full path to map yaml file to load"
+    )
+    declare_namespace_arg = DeclareLaunchArgument(
+        "namespace",
+        default_value=EnvironmentVariable("ROBOT_NAMESPACE", default_value=""),
+        description="Top-level namespace",
+    )
+    declare_observation_topic_arg = DeclareLaunchArgument(
+        "observation_topic",
+        default_value="",
+        description="Topic name for LaserScan or PointCloud2 observation messages type.",
+    )
+    declare_observation_topic_type_arg = DeclareLaunchArgument(
+        "observation_topic_type",
+        default_value="pointcloud",
+        description="Observation topic type.",
+        choices=["laserscan", "pointcloud"],
+    )
+    declare_params_file_arg = DeclareLaunchArgument(
+        "params_file",
+        default_value=PathJoinSubstitution(
+            [
+                husarion_ugv_navigation,
+                "config",
+                PythonExpression(["'nav2_", observation_topic_type, "_params.yaml'"]),
```

Comment:
Actually is this suggestion work the same?
```suggestion
               ["'nav2_", observation_topic_type, "_params.yaml'"],
```

Reply:
I merge config into one `nav2_params.yaml`. Please check again.